### ISSUE TITLE
Breakdown metrics are being sent even if breakdownMetrics is set to false

### ISSUE THREAD

**Issue by AE777F63** [elastic/apm-agent-nodejs#1460]
**Describe the bug**

Breakdown metrics are being sent even if breakdownMetrics is set to false.

**To Reproduce**

1. Start APM with breakdownMetrics: false
```
apm.start({
    ...
    breakdownMetrics: false
});
```
2. Send some requests to the server
3. Wait until APM sends the metrics report
4. Check the collected metrics in Kibana and see events like this being reported:
```
{
  "_index": "apm-7.4.0-metric-2019.10.20",
  "_type": "_doc",
  "_id": "...",
  "_score": 1,
  "_source": {
    "process": {
      ...
    },
    "agent": {
      "name": "nodejs",
      "version": "3.1.0"
    },
    "processor": {
      "name": "metric",
      "event": "metric"
    },
    "labels": {
      ...
    },
    "observer": {
      "hostname": "...",
      "id": "...",
      "ephemeral_id": "...",
      "type": "apm-server",
      "version": "7.4.0",
      "version_major": 7
    },
    "@timestamp": "2019-10-20T18:56:04.644Z",
    "ecs": {
      "version": "1.1.0"
    },
    "service": {
      ...
    },
    "host": {
      ...
    },
    "transaction": {
      "duration": {
        "count": 2,
        "sum": {
          "us": 659.641
        }
      },
      "breakdown": {
        "count": 2
      },
      "name": "GET unknown route",
      "type": "request"
    }
  },
  "fields": {
    "@timestamp": [
      "2019-10-20T18:56:04.644Z"
    ]
  }
}
```

**Expected behavior**

Breakdown events should not be reported if disabled via configuration.

**Environment (please complete the following information)**

- Node.js version: 10.16.3
- APM Server version: 7.4.0
- Agent version: 3.1.0

**How are you starting the agent? (please tick one of the boxes)**

- [x] Calling `agent.start()` directly (e.g. `require('elastic-apm-node').start(...)`)
- [ ] Requiring `elastic-apm-node/start` from within the source code
- [ ] Starting node with `-r elastic-apm-node/start`

**Additional context**

Workaround:
```
const Transaction = require('elastic-apm-node/lib/instrumentation/transaction');
Transaction.prototype._captureBreakdown = () => null;
```

**Comment by Qard** [comment]
The `breakdownMetrics` config is only supposed to prevent reporting of _span_ metricsets, not _transaction_ metricsets. It _should_ be omitting the `transaction.breakdown.count` part though. The `transaction.duration.*` data is not part of the breakdown metrics data, it's just reported together because the label sets happen to be the same.

If you want to prevent the other data being sent, I'm afraid your workaround is currently the only way. We've got an issue for a [`disabledMetrics` string filter](https://github.com/elastic/apm-agent-nodejs/issues/902), but no particular prioritization on it.

**Comment by AE777F63** [comment]
OK, I'll stick to the workaround then.
I guess a cleaner solution would be to remove those metrics in a filter, but apparently filters are only applied to spans and transactions.

**Comment by Qard** [comment]
Indeed, there's no metricset filter yet. It shouldn't be _too_ terribly complicated to implement though. PRs of course welcome. I've created a new issue for filter functions here: https://github.com/elastic/apm-agent-nodejs/issues/1469

**Comment by Qard** [comment]
Closing this as I think it's resolved by #1466 and #1469. Let me know if you disagree and I can reopen this.
